tasks/ast_tools/src/generators/typescript.rs

```diff
@@ -0,0 +1,170 @@
+use convert_case::{Case, Casing};
+use itertools::Itertools;
+use std::{
+    io::Write,
+    process::{Command, Stdio},
+};
+
+use super::define_generator;
+use crate::{
+    codegen::LateCtx,
+    output,
+    schema::{
+        serialize::{enum_variant_name, get_type_tag},
+        EnumDef, GetIdent, StructDef, TypeDef, TypeName,
+    },
+    Generator, GeneratorOutput,
+};
+
+const CUSTOM_TYPESCRIPT: &str = include_str!("../../../../crates/oxc_ast/src/ast/types.d.ts");
+
+define_generator! {
+    pub struct TypescriptGenerator;
+}
+
+impl Generator for TypescriptGenerator {
+    fn generate(&mut self, ctx: &LateCtx) -> GeneratorOutput {
+        let file = file!().replace('\\', "/");
+        let mut content = format!(
+            "\
+        		// To edit this generated file you have to edit `{file}`\n\
+        		// Auto-generated code, DO NOT EDIT DIRECTLY!\n\n\
+						{CUSTOM_TYPESCRIPT}\n\
+						"
```

Comment:
Nit: Weird indentation.

Reply:
Ah I think it's just a tabs-vs-spaces thing. Probably some tabs have crept in, so it renders weirdly if you have your IDE set to render tabs as 4 spaces. Our convention is to use spaces instead of tabs.